src/qb4-packw/gen/qb4-packw-x16c4-gemm-goi-neondot.c

```diff
@@ -0,0 +1,405 @@
+// clang-format off
+// Auto-generated file. Do not edit!
+//   Template: src/qb4-packw/kr-neondot.c.in
+//   Generator: tools/xngen
+//
+// Copyright 2023 Google LLC
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree.
+
+
+#include <assert.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <math.h>
+#include <arm_neon.h>
+
+#include "src/xnnpack/packw.h"
+
+// convert a vector from packed nibbles to planar, and accumulate sum
+static XNN_INTRINSIC
+int8x16_t xnn_packed2planar(
+  int32x4_t *vacc,
+  const uint8x16_t v,
+  const uint8x16_t vmask,
+  const uint8x16_t veor_mask,
+  const int32x4_t neg_zp)
+{
+  const uint8x16_t vl = vshrq_n_u8(v, 4);  // isolate lower int 4
+  const uint8x16_t vh = vandq_u8(v, vmask);  // isolate upper int 4
+  const uint32x4_t vh_acc = vpaddlq_u16(vpaddlq_u8(vh));
+  const uint32x4_t vl_acc = vpaddlq_u16(vpaddlq_u8(vl));
```

Comment:
use dotprod with zero here

Reply:
ahh oops, i had this in another branch but it failed to get in here. Let me update